core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -2165,8 +2166,15 @@ private void analyzeFiltersAndMasks(Table table, QualifiedObjectName name, Optio
                 }
             }
 
-            accessControl.getRowFilters(session.toSecurityContext(), name)
-                    .forEach(filter -> analyzeRowFilter(session.getIdentity().getUser(), table, name, accessControlScope, filter));
+            for (ViewExpression filter : accessControl.getRowFilters(session.toSecurityContext(), name)) {
+                Expression expression = analyzeRowFilter(session.getIdentity().getUser(), table, name, accessControlScope, filter);
+                analysis.addRowFilter(table, expression);
+            }
+            for (String checkConstraint : checkConstraints) {
+                ViewExpression filter = new ViewExpression(session.getIdentity().getUser(), Optional.of(name.getCatalogName()), Optional.of(name.getSchemaName()), checkConstraint);
+                Expression expression = analyzeRowFilter(session.getIdentity().getUser(), table, name, accessControlScope, filter);
```

Comment:
`analyzeRowFilter` technically does more or less what we want -- it takes a `String` expression and produces an `Expression`.
However, the method name and checks done in that method (e.g. `analysis.hasRowFilter(name, currentIdentity)` or `analysis.registerTableForRowFiltering(name, currentIdentity)`) are not applicable to check constraints.

Let's rollback changes to `analyzeRowFilter`.
Let's extract common code for converting connector-provided `String` into `Expression` (with due verifications).